### PR TITLE
871858 - [UG Doc] Prepare the UG documentation for the pull to refresh support

### PR DESCRIPTION
**Pull request by SethupathyD**
### Description.
 Prepare the UG documentation for the pull to refresh and column sizer helper method support.

### CODE REVIEW

File: MAUI/DataGrid/pull-to-refresh.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: PullToRefresh in .NET MAUI DataGrid control | Syncfusion
+description: Learn here all about PullToRefresh support in Syncfusion .NET MAUI DataGrid (SfDataGrid) control and more.
+platform: MAUI
+control: SfDataGrid
+documentation: UG
+---
+
+# PullToRefresh in MAUI DataGrid (SfDataGrid)
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:10]: Pull To Refresh in MAUI DataGrid (SfDataGrid)

File: MAUI/DataGrid/pull-to-refresh.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: PullToRefresh in .NET MAUI DataGrid control | Syncfusion
+description: Learn here all about PullToRefresh support in Syncfusion .NET MAUI DataGrid (SfDataGrid) control and more.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:4]: Learn here all about the pull to refresh support in Syncfusion .NET MAUI DataGrid (SfDataGrid) control and more.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
+---
+layout: post
+title: PullToRefresh in .NET MAUI DataGrid control | Syncfusion
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:3]: title: Pull To Refresh in .NET MAUI DataGrid Control | Syncfusion

File: MAUI/DataGrid/pull-to-refresh.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                this.viewModel!.ItemsSourceRefresh();
+                this.pullToRefresh.IsRefreshing = false;
+            }
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## TransitionMode
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:134]: ## Transition mode

File: MAUI/DataGrid/pull-to-refresh.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+The [TransitionMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.PullToRefresh.SfPullToRefresh.html#Syncfusion_Maui_PullToRefresh_SfPullToRefresh_TransitionMode) property specifies the mode of the animations. It has the following two modes:
+
+* [SlideOnTop](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.PullToRefresh.PullToRefreshTransitionType.html#Syncfusion_Maui_PullToRefresh_PullToRefreshTransitionType_SlideOnTop)
+* [Push](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.PullToRefresh.PullToRefreshTransitionType.html#Syncfusion_Maui_PullToRefresh_PullToRefreshTransitionType_Push)
+
+The default transition is `SlideOnTop` that draws the RefreshView on top of the `PullableContent`.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:141]: The default transition is `SlideOnTop` that draws the RefreshView on top of the `DataGrid`

File: MAUI/DataGrid/pull-to-refresh.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+{% tabs %}
+{% highlight xaml tabtitle="MainPage.xaml" hl_lines="2" %}
+
+    <syncfusion:SfPullToRefresh x:Name="pullToRefresh"
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:146]: Provide code with pullable content of DataGrid.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+{% tabs %}
+{% highlight xaml tabtitle="MainPage.xaml" hl_lines="2" %}
+
+    <syncfusion:SfPullToRefresh x:Name="pullToRefresh" 
+                                TransitionMode="SlideOnTop" />
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+    pullToRefresh.TransitionMode = PullToRefreshTransitionType.SlideOnTop;
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:152]: Xaml code alone is enough.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+{% endhighlight %}
+{% endtabs %}
+
+The following code example shows how to set the `TransitionMode` as `Push` to PullToRefresh. This transition moves only the refresh content and the header is fixed.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:157]: The following code example shows how to set the `TransitionMode` as `Push` to `SfPullToRefresh`. This transition moves only the refresh content and the header is fixed.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+{% tabs %}
+{% highlight xaml tabtitle="MainPage.xaml" hl_lines="2" %}
+
+    <syncfusion:SfPullToRefresh x:Name=" pullToRefresh"
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:162]: Provide code with pullable content of DataGrid.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+{% tabs %}
+{% highlight xaml tabtitle="MainPage.xaml" hl_lines="2" %}
+
+    <syncfusion:SfPullToRefresh x:Name=" pullToRefresh" 
+                                TransitionMode="Push" />
+
+{% endhighlight %}
+{% highlight c# tabtitle="MainPage.xaml.cs" %}
+
+    pullToRefresh.TransitionMode = PullToRefreshTransitionType.Push;
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:168]: Xaml code alone is enough.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+title: PullToRefresh in .NET MAUI DataGrid control | Syncfusion
+description: Learn here all about PullToRefresh support in Syncfusion .NET MAUI DataGrid (SfDataGrid) control and more.
+platform: MAUI
+control: SfDataGrid
+documentation: UG
+---
+
+# PullToRefresh in MAUI DataGrid (SfDataGrid)
+
+PullToRefresh is a refresh control that allows you to interact and refresh the view loaded in it. The PullToRefresh control allows you to refresh the view upon performing the pull-to-refresh action. A progress indicator will be shown while starting the pulling. The application will be refreshed once you have pulled down a certain distance and released the touch.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:12]: The above description is not required. Check and remove.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+{% endhighlight %}
+{% endtabs %}
+    <br/>
+    <li> Define the DataGrid as PullableContent of the PullToRefresh.</li> 
+    <li> Handle the pull to refresh events for refreshing the data. </li>
+    <li> Customize the required properties of the DataGrid and PullToRefresh based on your requirement.</li>
+</ol>
+
+This is how the final output will look like when hosting a Datagrid control as pullable content.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:41]: This is how the final output will look like when hosting a DataGrid control as pullable content.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/pull-to-refresh.md (hunk 3 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:3]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 2 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:4]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 1 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:10]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 11 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:12]: image
Had removed as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 12 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:41]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 4 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:134]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 5 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:141]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 6 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:146]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 7 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:152]: image
Had removed the .cs code.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 8 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:157]: image
Had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 9 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:162]: image
had updated as discussed.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 10 above)
Comment (SethupathyD) [reply MAUI/DataGrid/pull-to-refresh.md:168]: image
Had removed the .cs code.

File: MAUI/DataGrid/overview.md [hunk 13]
```diff
@@ -47,6 +47,8 @@ The .NET MAUI DataGrid control is used to display and manipulate data in a tabul
 
 * **Load More** -  Display an interactive view when the grid reaches its maximum or minimum offset while scrolling down or up. Tapping the interactive view triggers a command to add more data to the item source of the grid at runtime.
 
+* **Pull To Refresh** - Support enhances the user experience by enabling data refreshing through a pull-down gesture. This feature is particularly useful when dealing with large datasets or real-time data updates in mobile or desktop applications.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/overview.md:50]: Support allows users to load a subset of data when the DataGrid is pulled down.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/overview.md (hunk 13 above)
Comment (SethupathyD) [reply MAUI/DataGrid/overview.md:50]: image
Had updated as discussed.

**Review by ashok-kuvaraja — APPROVED** [review]
The code changes are fine.

**Pull request merged**